src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -40,26 +41,57 @@ Function: java_bytecode_languaget::get_language_options
 
 \*******************************************************************/
 
-void java_bytecode_languaget::get_language_options(const cmdlinet &cmd)
+void java_bytecode_languaget::get_language_options(
+  const cmdlinet &cmd, optionst &options)
 {
-  disable_runtime_checks=cmd.isset("disable-runtime-check");
-  assume_inputs_non_null=cmd.isset("java-assume-inputs-non-null");
-  string_refinement_enabled=cmd.isset("refine-strings");
+  options.set_option(
+    "disable-runtime-check",
+    disable_runtime_checks=cmd.isset("disable-runtime-check"));
+
+  options.set_option(
+    "java-assume-inputs-non-null",
+    assume_inputs_non_null=cmd.isset("java-assume-inputs-non-null"));
+
+  options.set_option(
+    "refine-strings",
+    string_refinement_enabled=cmd.isset("refine-strings"));
+
```

Comment:
Option parsing from cmdlinet to optiont should be done with the rest of the parse_option code (in the application directory), not in the front-end.